backend/src/test/java/org/uresti/pozarreal/service/impl/HousesServiceImplTests.java

```diff
@@ -0,0 +1,223 @@
+        // Given:
+        HousesRepository housesRepository = Mockito.mock(HousesRepository.class);
+        StreetRepository streetRepository = Mockito.mock(StreetRepository.class);
+        RepresentativeRepository representativeRepository = Mockito.mock(RepresentativeRepository.class);
+        SessionHelper mockSessionHelper = Mockito.mock(SessionHelper.class);
+
+        HousesServiceImpl housesService = new HousesServiceImpl(housesRepository, streetRepository, representativeRepository, mockSessionHelper);
+
+        String userId = UUID.randomUUID().toString();
+        LoggedUser user = LoggedUser.builder().claims(Collections.singletonMap("userId", userId)).build();
+
+        org.uresti.pozarreal.model.Representative representative = org.uresti.pozarreal.model.Representative.builder()
+                .street("Street 2")
+                .house("house 1")
+                .phone("123456")
+                .userId("userId1")
+                .build();
+
+        Mockito.when(mockSessionHelper.hasRole(user, Role.ROLE_REPRESENTATIVE)).thenReturn(true);
+        Mockito.when(mockSessionHelper.hasRole(user, Role.ROLE_ADMIN)).thenReturn(false);
+        Mockito.when(representativeRepository.findById(userId)).thenReturn(java.util.Optional.of(representative));
+
+        // When:
+        // Then:
+        Assertions.assertThatThrownBy(() -> housesService.getHousesByStreet("Street 1", user))
+                .isInstanceOf(BadRequestDataException.class)
+                .hasMessage("Invalid street for representative query", "INVALID_STREET");
+    }
+
+    @Test
+    public void givenAnHouseId_thenToggleChipStatusRequest() {
+        // Given:
+        HousesRepository housesRepository = Mockito.mock(HousesRepository.class);
+        StreetRepository streetRepository = Mockito.mock(StreetRepository.class);
+        RepresentativeRepository representativeRepository = Mockito.mock(RepresentativeRepository.class);
+        SessionHelper mockSessionHelper = Mockito.mock(SessionHelper.class);
+        HousesServiceImpl housesService = new HousesServiceImpl(housesRepository, streetRepository, representativeRepository, mockSessionHelper);
+
+        org.uresti.pozarreal.model.House house = new org.uresti.pozarreal.model.House();
+        house.setId("1");
+        house.setNotes("hello");
+        house.setStreet("Street 1");
+        house.setNumber("123456");
+        house.setChipsEnabled(true);
+
+        Street street = new Street();
+
+        street.setId("1");
+        street.setName("name 1");
+
+        Mockito.when(housesRepository.save(house)).thenReturn(house);
+        Mockito.when(streetRepository.findById(house.getStreet())).thenReturn(java.util.Optional.of(street));
+        Mockito.when(housesRepository.findById("1")).thenReturn(java.util.Optional.of(house));
+
+        // When:
+        housesService.toggleChipStatusRequest("1", false);
```

Comment:
Test only one of the methods ... call getHouseInfo doesn't help to test toggleChipStatusRequest as you don't really use a real repository, it is a mocked one. 

In order to test toggleChipStatusRequest it is enough to test interactions

Reply:
done